kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -279,13 +280,120 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    /// Create an initializer for a generator struct.
+    fn codegen_rvalue_generator(&mut self, operands: &[Operand<'tcx>], ty: Ty<'tcx>) -> Expr {
+        let layout = self.layout_of(ty);
+        let discriminant_field = match &layout.variants {
+            Variants::Multiple { tag_encoding: TagEncoding::Direct, tag_field, .. } => tag_field,
+            _ => unreachable!("Generators have more than one variant and use direct encoding"),
```

Comment:
Can you please add the layout to this message? Something like:
```rust
unreachable!("Expected generators to have multiple variants and direct encoding, but found: {layout:?}")
```

Reply:
Done, thank you for the suggestion!